### PR TITLE
Rtc rv8523

### PR DESCRIPTION
**Pull request by JonBruchim**

### CODE REVIEW

File: drivers/rv8523_rtc.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            self._pig.i2c_write_device(self._dev, [reg])
+        read_size, clock_unit_bcd = self._pig.i2c_read_device(self._dev, 1)
+
+        if read_size == 1:
+            return self.bcd_to_int(clock_unit_bcd & mask)
+
+        return self.ERR_FAIL
+
+    def _get_time(self):
+        seconds = self._get_clock_unit(0x7F, self.REG_SECONDS)
```
Comment (JonBruchim) [inline drivers/rv8523_rtc.py:114]: will change these magics later today

File: drivers/rtc_test.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from rv8523_rtc import Rv8523Rtc
```
Comment (osherdp) [inline drivers/rtc_test.py:1]: move this file to `drivers/debug`

File: drivers/rv8523_rtc.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            self._pig.i2c_write_device(self._dev, ctrl_1)
+        except pigpio.error as e:
+            log.error("Could not write control 1 reg to RTC. "
+                      "Is the RTC connected?.")
+            raise I2CWriteError("i2c write failed")
+
+    def bcd_to_int(self, bcd):
+        units = bcd & self.BCD_NIBBLE_MASK
+        tens = (bcd >> self.BCD_TENS_SHIFT) & self.BCD_NIBBLE_MASK
+        return (tens * 10 + units)
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:88]: remove redundant parenthesis
here and elsewhere

File: drivers/rv8523_rtc.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+    @staticmethod
+    def set_system_time(dt):
+        second = str(dt.second)
+        minute = str(dt.minute)
+        hour = str(dt.hour)
+        day = str(dt.day)
+        month = str(dt.month)
+        year = str(dt.year)
+        subprocess.call("sudo date -s '" + year + "-" + month + "-" + day +
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:158]: use f-string

File: drivers/rv8523_rtc.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        try:
+            self._set_clock_unit(date.second, self.REG_SECONDS)
+            self._set_clock_unit(date.minute)
+            self._set_clock_unit(date.hour)
+            self._set_clock_unit(date.day)
+            self._set_clock_unit(date.month, self.REG_MONTHS)
+            self._set_clock_unit(date.year - self.REG_YEARS_OFFSET)
+        except pigpio.error as e:
+            log.error("Could not set RTC time. "
+                      "Is the RTC connected?.")
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:133]: remove dots all over the place

File: drivers/rv8523_rtc.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                return time
+            else:
+                log.error("rtc read error")
+                raise I2CReadError("RTC get time unavailable.")
+        except pigpio.error as e:
+            log.error("Could not read from RTC. "
+                      "Is the RTC connected?.")
+            raise I2CReadError("i2c write failed")
+
+    @staticmethod
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:150]: why staticmethod?

Comment (JonBruchim) [reply drivers/rv8523_rtc.py:149]: Doesn't depend on the driver instance itself.
Anyone could use it

File: drivers/rv8523_rtc.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    REG_YEARS_OFFSET = 2000
+
+    def __init__(self):
+        super().__init__()
+
+        # Start RTC
+        try:
+            self._rtc_start()
+        except pigpio.error as e:
+            raise e
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:35]: If you're just raising the same exception, delete the `try-except`
or raise a different exception if it's needed

File: drivers/rv8523_rtc.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+        log.debug("rtc initialized")
+
+    def set_24h_mode(self):
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:39]: unused?

File: drivers/rv8523_rtc.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        units = number % 10
+        tens = int(number / 10) << self.BCD_TENS_SHIFT
+        return tens + units
+
+    def _set_clock_unit(self, value, reg):
+        try:
+            value = self.int_to_bcd(value)
+            self._pig.i2c_write_device(self._dev, [reg, value])
+        except pigpio.error as e:
+            raise e
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:83]: same as above

File: drivers/rv8523_rtc.py [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+from datetime import datetime
+import subprocess
+
+from errors import I2CReadError, I2CWriteError
+from .i2c_driver import I2cDriver
+
+log = logging.getLogger(__name__)
+
+
+class Rv8523Rtc(I2cDriver):
```
Comment (osherdp) [inline drivers/rv8523_rtc.py:12]: I think you need to define `close()` as well, even an empty one for the cleanup to be OK

File: rasp_init/install-as-service.sh [hunk 11]
```diff
@@ -20,6 +20,10 @@ echo "enabling pigpiod"
 sudo systemctl enable pigpiod
 echo "done!"
 
+echo "disable NTP time, get RTC time"
```
Comment (osherdp) [inline rasp_init/install-as-service.sh:23]:
```suggestion
echo "disable NTP time, use time from RTC"
```

**Pull request merged**